### PR TITLE
[RFR] Upgrade to MUI beta 24

### PR DESCRIPTION
**Pull request by djhi**
- [x] `Sidebar`, `Notification` and `FilterButton` : `OnRequestClose` to `onClose`
- [x] `Toolbar`: `noGutter` to `disableGutters`
- [x] Different handling of breakpoints between MUI `Hidden` component and our `Responsive` component => `AppBarMobile` and the main `Header` are both displayed on small devices
- [x] Typography misused (warning in both console and tests)

### CODE REVIEW

File: packages/react-admin/src/mui/layout/AppBarMobile.js [hunk 1]
```diff
@@ -33,39 +33,29 @@ const styles = {
     },
 };
 
-class AppBarMobile extends Component {
-    handleLeftIconButtonTouchTap = event => {
```
Comment (fzaninotto) [inline packages/react-admin/src/mui/layout/AppBarMobile.js]: weird that this wasn't used anymore...

File: packages/react-admin/src/mui/layout/Layout.js [hunk 2]
```diff
@@ -83,7 +83,7 @@ const Layout = ({
     <MuiThemeProvider theme={createMuiTheme(theme)}>
         <div className={classnames(classes.root, className)} {...rest}>
             <div className={classes.appFrame}>
-                <Hidden xsDown>
+                <Hidden smDown>
```
Comment (fzaninotto) [inline packages/react-admin/src/mui/layout/Layout.js]: needs more thorough tests ; unless something changed on the mui side, this looks odd.

Comment (djhi) [reply packages/react-admin/src/mui/layout/Layout.js]: Something changed. It's not inclusive anymore: https://github.com/mui-org/material-ui/releases/tag/v1.0.0-beta.24

File: packages/react-admin/src/mui/layout/AppBarMobile.js (hunk 1 above)
Comment (djhi) [reply packages/react-admin/src/mui/layout/AppBarMobile.js]: Agreed, but it's not and the button is working

**Pull request merged**